Framework/Algorithms/test/ConvertTableToMatrixWorkspaceTest.h

```diff
@@ -139,6 +139,88 @@ class ConvertTableToMatrixWorkspaceTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnX", "A"));
     TS_ASSERT_THROWS_NOTHING(m_converter->setPropertyValue("ColumnY", "B"));
   }
+  // test if it can convert a string to a double, if the string is numeric
+  void testStringToDouble() {
+
+    ITableWorkspace_sptr tws = WorkspaceFactory::Instance().createTable();
+    tws->addColumn("str", "A");
+    tws->addColumn("double", "B");
+    tws->addColumn("double", "C");
+
+    size_t n = 10;
+    for (size_t i = 0; i < n; ++i) {
+      TableRow row = tws->appendRow();
```

Comment:
Ok I had a look. `appendRow` actually returns a `TableRowHelper` object. `TableRow` has a constructor ` TableRow(const TableRowHelper &trh)` hence we have an assignment initialization here. So yes, having `auto` there would give a compilation error because the stream operators are not overloaded for the `TableRowHelper` type.